Src/VimCore/ExternalEdit.fs

```diff
@@ -7,7 +7,7 @@ type internal ExternalEditMode(_vimBufferData: IVimBufferData) =
     member x.ClearSelection () =
         let textView = _vimBufferData.TextView
         let start = textView.Selection.Start
-        textView.Selection.Clear()

```

Comment:
Random musing: This is where Roslyn analyzers would be nice for F#. Effectively we need to ban `ITextSelection.Clear` in VsVim now. Can't think of a good way to do this in F# except for diligent code review.

Reply:
We have a similar situation with several other APIs, notably `ITextView.TextViewLines`, etc. The situation is not as bad you might think because 90% of the time, some existing code is used as model for how to do implement some new code and then the same API gets used because the old code followed the model so the new code also follows the model.

My policy is that code review enforces consistency but the tests enforce correctness. So if someone doesn't use the supplied API but there is no bug as a consequence, it's just inconsistent, not wrong!